yt/loaders.py

```diff
@@ -1457,3 +1463,108 @@ def load_sample(
         loadable_path = loadable_path.joinpath(load_name, specific_file)
 
     return load(loadable_path, **kwargs)
+
+
+# --- Loader for tar-based datasets ---
+def load_archive(
+    fn: Union[str, Path],
+    path: str,
+    ratarmount_kwa: Optional[Dict] = None,
+    *args,
+    **kwargs,
+) -> Dataset:
+    r"""
+    Load archived data with yt.
+
+    This is a wrapper around :func:`~yt.loaders.load` to include mounting
+    and unmounting the archive as a read-only filesystem and load it.
+
+    Parameters
+    ----------
+
+    fn: str
+        The `filename` of the archive containing the dataset.
+
+    path: str
+        The path to the dataset in the archive.
+
+    ratarmount_kwa: dict, optional
+        Optional parameters to pass to ratarmount to mount the archive.
+
+    Notes
+    -----
+
+    - Any additional keyword argument is passed down to :func:`~yt.loaders.load`.
+    - This function requires ratarmount.
+    - This function does not work on Windows system.
+    """
+    if platform.system() not in ("Linux", "Darwin"):
+        raise RuntimeError(
+            "Reading from an archive is not supported on your operating system"
+            f" ({platform.system()})."
+        )
```

Comment:
I assume this limitation is really coming from ratarmount, in which case we shouldn't add another barrier on top of it, in case a future version has support for windows. I would suggest to wrap the first call to ratarmount that may break in a try statement and possibly raise a custom error message (as you're doing here), but be careful about using a conditional (if it fails, it's _probably_ because of platform support, but we don't know it for a fact).